server/api/api.go

```diff
@@ -35,33 +35,53 @@ type handler struct {
 
 type handlerWithUserID func(w http.ResponseWriter, r *http.Request, userID string)
 
-func newHandler(sp splunk.Splunk) *handler {
+func newHandler(sp splunk.Splunk, c *config.Config) *handler {
 	h := &handler{
 		Router: mux.NewRouter(),
 		sp:     sp,
 	}
 	apiRouter := h.Router.PathPrefix(config.APIPath).Subrouter()
 
-	apiRouter.HandleFunc(WebhookEndpoint, h.handleAlertActionWH()).Methods(http.MethodPost)
+	apiRouter.HandleFunc(WebhookEndpoint, h.handleAlertActionWH(c)).Methods(http.MethodPost)
 
 	return h
 }
 
-func (h *handler) handleAlertActionWH() http.HandlerFunc {
+func (h *handler) handleAlertActionWH(config *config.Config) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		var req splunk.AlertActionWHPayload
 		err := json.NewDecoder(r.Body).Decode(&req)
 		if err != nil {
+			h.sp.LogError("Bad Request", "error", err)
 			h.jsonError(w, Error{Message: "Bad Request", StatusCode: http.StatusBadRequest})
 			return
 		}
-
-		id, err := getURLParam(r, "id")
+		id := r.URL.Query().Get("id")
+		if id == "" {
+			errMsg := "Bad webhook request. Missing url param 'id'"
+			h.sp.LogError(errMsg)
+			h.jsonError(w, Error{Message: errMsg, StatusCode: http.StatusBadRequest})
+		}
+		secret := r.URL.Query().Get("secret")
```

Comment:
To make this function more readable, can we add a new line _after_ each if block?

```suggestion
		id := r.URL.Query().Get("id")
		if id == "" {
			errMsg := "Bad webhook request. Missing url param 'id'"
			h.sp.LogError(errMsg)
			h.jsonError(w, Error{Message: errMsg, StatusCode: http.StatusBadRequest})
		}
// new line right here
		secret := r.URL.Query().Get("secret")
```